pkg/reconcile/pipeline/infinispan/handler/manage/upgrades.go

```diff
@@ -185,12 +179,17 @@ func GracefulShutdown(i *ispnv1.Infinispan, ctx pipeline.Context) {
 			ctx.Requeue(fmt.Errorf("Spec.Replicas(%d) must be 0 or equal to Status.ReplicasWantedAtRestart(%d)", i.Spec.Replicas, i.Status.ReplicasWantedAtRestart))
 			return
 		}
+
 		ctx.Requeue(
 			ctx.UpdateInfinispan(func() {
+				i.Status.Replicas = &i.Spec.Replicas
 				i.Status.ReplicasWantedAtRestart = 0
 				i.SetCondition(ispnv1.ConditionGracefulShutdown, metav1.ConditionFalse, "")
+				// Add ConditionScalingUp so that on requeue we update the ClusterService so that no pods will be selected
+				i.SetCondition(ispnv1.ConditionScalingUp, metav1.ConditionTrue, "")
 			}),
 		)
+		return
```

Comment:
I guess this is intentional

Reply:
That's a leftover from an earlier approach that I implemented, I'll remove.